controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -198,23 +199,23 @@ func (r *Reconciler) deleteNSTemplateSet(logger logr.Logger, nsTmplSet *toolchai
 // deleteRedundantObjects takes template objects of the current tier and of the new tier (provided as newObjects param),
 // compares their names and GVKs and deletes those ones that are in the current template but are not found in the new one.
 // return `true, nil` if an object was deleted, `false, nil`/`false, err` otherwise
-func deleteRedundantObjects(logger logr.Logger, client client.Client, deleteOnlyOne bool, currentObjs []applycl.ToolchainObject, newObjects []applycl.ToolchainObject) (bool, error) {
+func deleteRedundantObjects(logger logr.Logger, client client.Client, deleteOnlyOne bool, currentObjs []runtimeclient.Object, newObjects []runtimeclient.Object) (bool, error) {
 	deleted := false
 	logger.Info("checking redundant objects", "count", len(currentObjs))
 Current:
 	for _, currentObj := range currentObjs {
-		logger.Info("checking redundant object", "objectName", currentObj.GetGvk().Kind+"/"+currentObj.GetName())
+		logger.Info("checking redundant object", "objectName", currentObj.GetObjectKind().GroupVersionKind().Kind+"/"+currentObj.GetName())
 		for _, newObj := range newObjects {
-			if currentObj.HasSameGvkAndName(newObj) {
+			if currentObj.GetObjectKind().GroupVersionKind() == newObj.GetObjectKind().GroupVersionKind() && currentObj.GetName() == newObj.GetName() {
```

Comment:
wouldn't it be nice to create a helper function called `HasSameGvkAndName` taking two parameters (currentObj & newObj) and doing this comparison and call it here? It would be easier to read.

Reply:
ok, let me add that in https://github.com/codeready-toolchain/toolchain-common/pull/211

Reply:
done in 578ddf7